pkg/api/common_test.go

```diff
@@ -411,6 +412,9 @@ func setupHTTPServerWithCfgDb(t *testing.T, useFakeAccessControl, enableAccessCo
 		initCtx.Context = c
 		initCtx.Logger = log.New("api-test")
 		c.Map(initCtx)
+
+		c.Req = c.Req.WithContext(ctxkey.Set(c.Req.Context(), initCtx))
+		c.Map(c.Req)
```

Comment:
Because tests mess with the internals of `pkg/web`, we need to do this several times here.

we could possibly modify `Mux.Map()` so that it detects `models.ReqContext` above and does the mapping automatically.

I dislike both though. Opinions @zserge @sakjur?